source/MRViewer/MRSpaceMouseHandlerWindows.cpp

```diff
@@ -118,65 +123,70 @@ void SpaceMouseHandlerWindows::handle()
     if ( !initialized_ || joystickIndex_ == -1 )
         return;
 
-
+    auto& viewer = getViewerInstance();
     int count;
     const float* axesNew = glfwGetJoystickAxes( joystickIndex_, &count );
-    if ( count != 6 )
+    if ( count == 6 )
     {
-        spdlog::error( "Error SpaceMouseHandlerWindows : Wrong axes count" );
-        // TODO disconect device
-        assert( false );
-        return;
-    }
-    Vector3f translate( axesNew[0] - axes_[0], axesNew[1] - axes_[1], axesNew[2] - axes_[2] );
-    Vector3f rotate( axesNew[3] - axes_[3], axesNew[4] - axes_[4], axesNew[5] - axes_[5] );
-    std::copy( axesNew, axesNew + 6, axes_.begin() );
-    translate *= axesScale;
-    rotate *= axesScale;
-
-    auto& viewer = getViewerInstance();
-    if ( translate.lengthSq() > 1.e-3f || rotate.lengthSq() > 1.e-3f )
-        viewer.spaceMouseMove( translate, rotate );
+        if ( active_ )
+        {
+            Vector3f translate( axesNew[0] - axes_[0], axesNew[1] - axes_[1], axesNew[2] - axes_[2] );
+            Vector3f rotate( axesNew[3] - axes_[3], axesNew[4] - axes_[4], axesNew[5] - axes_[5] );
+            translate *= axesScale;
+            rotate *= axesScale;
 
+            if ( translate.lengthSq() > 1.e-3f || rotate.lengthSq() > 1.e-3f )
+                viewer.spaceMouseMove( translate, rotate );
+        }
+        std::copy( axesNew, axesNew + 6, axes_.begin() );
+    }
 
     const unsigned char* buttons = glfwGetJoystickButtons( joystickIndex_, &count );
-    if ( count != 2 && count != 15 && count != 31 )
-    {
-        spdlog::error( "Error SpaceMouseHandlerWindows : Wrong buttons count" );
-        // TODO disconect device
-        assert( false );
-        return;
-    }
-    const int* mapButtons = mapButtonsPro;
-    if ( count == 2 )
-        mapButtons = mapButtonsCompact;
-    else if ( count == 15 )
-        mapButtons = mapButtonsPro;
-    else
-        mapButtons = mapButtonsEnterprise;
-    for ( int i = 0; i < count; ++i )
+    
+    if ( count == 2 || count == 15 || count == 31 )
```

Comment:
what is these numbers, please add comments

Reply:
added comment